hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/file/CsvFileFormat.java

```diff
@@ -34,6 +35,7 @@
     public static final String FORMAT_CSV = "csv";
 
     private Class<T> clazz;
+    private List<String> stringArrayFieldList;
```

Comment:
Does it make sense to have `clazz` & `stringArrayFieldList` at the same time? Maybe, both should be final and we should have 2 constructors setting `clazz` in one and `stringArrayFieldList` in the other?

Reply:
I followed the style already in the class, with a mutable `clazz`.

Reply:
We originaly had a final clazz, changed it with the SQL file connector. now looking at all the places maybe we could go back to final clazz and always provide a class (either Map, instead of null), TreeNode for json etc. This way we could avoid the cases `if null then this class`..